### PR TITLE
Add Trino 401 release notes

### LINKED ISSUE #14628: Release notes for 401

### PR DESCRIPTION
**Pull request by colebow**
## Description

Assemble the release notes for the upcoming Trino 400 release. Fixes #14628 

## Non-technical explanation

N/A

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

## Verification for each pull request

Format: maintainer, PR/issue number, ✅ / ❌ rn ✅ / ❌ docs
✅ rn - release note added and verified, or assessed to be not necessary, set to ❌ rn before completion
✅ docs - need for docs assessed and merged, or assessed to be not necessary, set to ❌ docs before completion

## 13 October 2022 ✅ 

@ebyhr #14610 ✅ rn ✅ docs
@ebyhr #14535 ✅ rn ✅ docs
@ebyhr #14612 ✅ rn ✅ docs
@ebyhr #14573 ✅ rn ✅ docs
@ebyhr #13558 ✅ rn ✅ docs

## 14 October 2022 ✅ 

@ebyhr #14560 ✅ rn ✅ docs
@ebyhr #14630 ✅ rn ✅ docs
@sopel39 #14596 ✅ rn ✅ docs
@sopel39 #14633 ✅ rn ✅ docs
@Praveen2112 #14615 ✅ rn ✅ docs
@findepi #14506 ✅ rn ✅ docs
@raunaqmorarka #14595 ✅ rn ✅ docs
@hashhar #12104 ✅ rn ✅ docs
@findepi #14525 ✅ rn ✅ docs
@ebyhr #14647 ✅ rn ✅ docs
@ebyhr #14642 ✅ rn ✅ docs

## 15 October 2022 ✅ 

@findepi #14643 ✅ rn ✅ docs
@findepi #14593 ✅ rn ✅ docs
@findepi #14592 ✅ rn ✅ docs
@findepi #14644 ✅ rn ✅ docs
@raunaqmorarka #14385 ✅ rn ✅ docs

## 17 October 2022 ✅

@losipiuk #13608 ✅ rn ✅ docs
@losipiuk #14655 ✅ rn ✅ docs
@findepi #14383 ✅ rn ✅ docs
@findepi #14585 ✅ rn ✅ docs
@sopel39 #14640 ✅ rn ✅ docs
@hashhar #14224 ✅ rn ✅ docs
@sopel39 #14635 ✅ rn ✅ docs
@electrum #14442 ✅ rn ✅ docs
@electrum #13874 ✅ rn ✅ docs
@electrum #13830 ✅ rn ✅ docs
@electrum #14084 ✅ rn ✅ docs
@sopel39 #14665 ✅ rn ✅ docs
@findepi #14577 ✅ rn ✅ docs

## 18 October 2022 ✅  

@hashhar #13688 ✅ rn ✅ docs
@sopel39 #14542 ✅ rn ✅ docs
@sopel39 #14666 ✅ rn ✅ docs
@electrum #14650 ✅ rn ✅ docs

## 19 October 2022 ✅ 

@hashhar #14657 ✅ rn ✅ docs
@sopel39 #14638 ✅ rn ✅ docs
@ebyhr #14662 ✅ rn ✅ docs
@sopel39 #14684 ✅ rn ✅ docs
@electrum #14695 ✅ rn ✅ docs

## 20 October 2022 ✅ 

@sopel39 #14553 ✅ rn ✅ docs
@findepi #14678 ✅ rn ✅ docs
@sopel39 #14699 ✅ rn ✅ docs
@arhimondr #14205 ✅ rn ✅ docs

## 21 October 2022

@ebyhr #14687 ✅ rn ✅ docs
@findepi #14703 ✅ rn ✅ docs
@losipiuk #14707 ✅ rn ✅ docs
@sopel39 #14660 ✅ rn ✅ docs
@arhimondr #14668 ✅ rn ✅ docs
@martint #7734 ✅ rn ✅ docs

## 22 October 2022

@losipiuk #14445 ✅ rn ✅ docs
@losipiuk #14716 ✅ rn ✅ docs
@kokosing #13713 ✅ rn ✅ docs
@kokosing #14625 ✅ rn ✅ docs
@kokosing #14607 ✅ rn ✅ docs

## 24 October 2022

@findepi #14426 ✅ rn ❌ docs
@losipiuk #14714 ✅ rn ✅ docs
@phd3 #14463 ✅ rn ✅ docs
@martint #14438 ✅ rn ✅ docs
@kokosing #14702 ✅ rn ✅ docs
@electrum #14731 ✅ rn ✅ docs
@electrum #14505 ✅ rn ✅ docs
@electrum #14036 ✅ rn ✅ docs
@electrum #14551 ✅ rn ✅ docs
@martint #14671 ✅ rn ✅ docs
@electrum #14297 ✅ rn ✅ docs
@findepi #14723 ✅ rn ✅ docs
@findepi #12523 ✅ rn ✅ docs

## 25 October 2022 ✅

@findepi #14728 ✅ rn ✅ docs
@findepi #14724 ✅ rn ✅ docs
@sopel39 #14572 ✅ rn ✅ docs
@kasiafi #14115 ✅ rn ✅ docs
@losipiuk #14730 ✅ rn ✅ docs
@ebyhr #14086 ✅ rn ✅ docs
@ebyhr #14712 ✅ rn ✅ docs
@ebyhr #14646 ✅ rn ✅ docs
@ebyhr #14629 ✅ rn ✅ docs

## 26 October 2022

@ebyhr #14759 ✅ rn ✅ docs
@findepi #14742 ✅ rn ✅ docs
@ebyhr #14727 ✅ rn ✅ docs
@sopel39 #13988 ✅ rn ✅ docs
@findepi #14562 ✅ rn ✅ docs
@kokosing #14732 ✅ rn ✅ docs
@findepi #14744 ✅ rn ✅ docs
@findepi #14327 ✅ rn ✅ docs

### CODE REVIEW

File: docs/src/main/sphinx/release/release-401.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  `exchange.s3.path-style-access` configuration property. ({issue}`14655`)
+* Add output buffer utilization distribution to `EXPLAIN ANALYZE VERBOSE`. ({issue}`14596`)
+* Add operator blocked time distribution to `EXPLAIN ANALYZE VERBOSE`. ({issue}`14640`)
+
+## Delta Lake connector
+
+* Add support for writing to tables using [version 4 of the writer](https://docs.delta.io/latest/versioning.html#features-by-protocol-version).
+  This does not yet include support for [change data feeds](https://docs.delta.io/2.0.0/delta-change-data-feed.html)
+  or generated columns. ({issue}`14573 `)
+* Fix bug where executing a `COMMENT` or `ALTER TABLE ... ADD COLUMN` statement
```
Comment (martint) [inline docs/src/main/sphinx/release/release-401.md:16]: What does it mean to "change the writer and reader versions"?

File: docs/src/main/sphinx/release/release-401.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  would change the reader and writer versions. ({issue}`14611`)
+* Fix failure when listing tables from the Glue metastore and one of the Glue
```
Comment (martint) [inline docs/src/main/sphinx/release/release-401.md:18]: Second "Glue" is redundant. It's implied by the first part of the sentence.

File: docs/src/main/sphinx/release/release-401.md (hunk 1 above)
Comment (ebyhr) [reply docs/src/main/sphinx/release/release-401.md:16]: Previously, the connector overwrites `delta.minReaderVersion` and `delta.minWriterVersion` table properties with `1` and `2`.

File: docs/src/main/sphinx/release/release-401.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+* Add support for using path-style access for all requests to S3 when using
+  fault-tolerant execution with exchange spooling. This can be enabled with the
+  `exchange.s3.path-style-access` configuration property. ({issue}`14655`)
+* Add output buffer utilization distribution to `EXPLAIN ANALYZE VERBOSE`. ({issue}`14596`)
+* Add operator blocked time distribution to `EXPLAIN ANALYZE VERBOSE`. ({issue}`14640`)
+
+## Delta Lake connector
+
+* Add support for writing to tables using [version 4 of the writer](https://docs.delta.io/latest/versioning.html#features-by-protocol-version).
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-401.md:13]:
```suggestion
* Add support for writing to tables with [Delta Laker writer protocol version 4](https://docs.delta.io/latest/versioning.html#features-by-protocol-version).
```

File: docs/src/main/sphinx/release/release-401.md (hunk 1 above)
Comment (mosabua) [inline docs/src/main/sphinx/release/release-401.md:16]: Maybe 

* Avoid changing reader and writer version table properties as unrelated side effect of adding comments.

File: docs/src/main/sphinx/release/release-401.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  This does not yet include support for [change data feeds](https://docs.delta.io/2.0.0/delta-change-data-feed.html)
+  or generated columns. ({issue}`14573 `)
+* Fix bug where executing a `COMMENT` or `ALTER TABLE ... ADD COLUMN` statement
+  would change the reader and writer versions. ({issue}`14611`)
+* Fix failure when listing tables from the Glue metastore and one of the Glue
+  tables has no properties. ({issue}`14577`)
+
+## Hive connector
+
+* Fix failure when listing tables from the Glue metastore and one of the Glue
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-401.md]: remove second "Glue"

File: docs/src/main/sphinx/release/release-401.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  tables has no properties. ({issue}`14577`)
+
+## Iceberg connector
+
+* Improve performance of the `remove_orphan_files` table procedure. ({issue}`13691`)
+* Fix query failure when analyzing a table containing a column with a
+  non-lowercase name. ({issue}`14583`)
+* Fix failure when listing tables from the Glue metastore and one of the Glue
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-401.md:31]: second "Glue"

File: docs/src/main/sphinx/release/release-401.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  tables has no properties. ({issue}`14577`)
+
+## Kafka connector
+
+* Add support for configuring the prefix for internal column names using the
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-401.md:36]:
```suggestion
* Add support for configuring the prefix for internal column names with the
```

File: docs/src/main/sphinx/release/release-401.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+  `kafka.internal-column-prefix` catalog configuration property. The default
+  value is `_` to maintain current behavior. ({issue}`14224`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-401.md:38]:
```suggestion
  value is `_`. ({issue}`14224`)
```

File: docs/src/main/sphinx/release/release-401.md [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+## MongoDB connector
+
+* Add `query` table function for query pass-through to the connector. ({issue}`14535`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-401.md:42]:
```suggestion
* Add the `query` table function for query pass-through to the database. ({issue}`14535`)
```

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/release/release-401.md (hunk 1 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-401.md:16]: fair .. sorry .. I misread.. .. leave the statements then..

File: docs/src/main/sphinx/release/release-401.md (hunk 8 above)
Comment (colebow) [reply docs/src/main/sphinx/release/release-401.md:42]: I'm going to leave this as-is for consistency, we've done the release note this way for all of the other `query` functions that have been added.

File: docs/src/main/sphinx/release/release-401.md (hunk 7 above)
Comment (colebow) [reply docs/src/main/sphinx/release/release-401.md:38]: I'm not sure this is an improvement, I'd rather provide the context that you don't need to change it and current behavior is unchanged.

File: docs/src/main/sphinx/release/release-401.md [hunk 9]
```diff
@@ -5,23 +5,35 @@
...
   fault-tolerant execution with exchange spooling. This can be enabled with the
   `exchange.s3.path-style-access` configuration property. ({issue}`14655`)
+* Add support for polymorphic table functions in file-based access control. ({issue}`13713`)
 * Add distribution data for output buffer utilization to
   `EXPLAIN ANALYZE VERBOSE`. ({issue}`14596`)
 * Add distribution data for operator blocked time to `EXPLAIN ANALYZE VERBOSE`. ({issue}`14640`)
 * Improve performance and reliability of `INSERT`, `MERGE`, and table optimization. ({issue}`14553`)
+* Fix query failure caused by a
+  `com.google.common.base.VerifyException: cannot unset noMoreSplits` error. ({issue`14668`})
+* Fix underestimation of CPU usage and scheduled time statistics for the build
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-401.md:15]: Maybe 

Correct estimation of

File: docs/src/main/sphinx/release/release-401.md [hunk 10]
```diff
@@ -5,23 +5,35 @@
...
+  side of joins in `EXPLAIN ANALYZE`. ({issue}`14572`)
 
 ## Delta Lake connector
 
 * Add support for writing to tables with [Delta Laker writer protocol version 4](https://docs.delta.io/latest/versioning.html#features-by-protocol-version).
   This does not yet include support for [change data feeds](https://docs.delta.io/2.0.0/delta-change-data-feed.html)
   or generated columns. ({issue}`14573 `)
+* Add support for writes on Google Cloud Storage. ({issue}`12264`)
+* Add support for DML statements on tables backed by GCS
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-401.md:24]: DML ?? might have to detail that.. also GCS .. maybe spell out again or add in line above

File: docs/src/main/sphinx/release/release-401.md [hunk 11]
```diff
@@ -43,6 +55,19 @@
...
+
+* Add support for [fault-tolerant execution](#fault-tolerant-execution). ({issue}`14445`)
+
+## PostgreSQL connector
+
+* Add support for [fault-tolerant execution](#fault-tolerant-execution). ({issue}`14445`)
+
+## SQL Server connector
+
+* Add support for writes when [fault-tolerant
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-401.md:68]: Why is this different from the two above?

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/release/release-401.md (hunk 10 above)
Comment (colebow) [reply docs/src/main/sphinx/release/release-401.md:24]: The line above was the fix for it, I just forgot to delete. 😆

File: docs/src/main/sphinx/release/release-401.md (hunk 9 above)
Comment (colebow) [reply docs/src/main/sphinx/release/release-401.md:15]: I'd rather leave it as-is. I don't think that's really any clearer, and this maintains the pattern of using 'fix' as the first word to describe fixes.

Comment (mosabua) [reply docs/src/main/sphinx/release/release-401.md:15]: fine by me

File: docs/src/main/sphinx/release/release-401.md [hunk 12]
```diff
@@ -0,0 +1,74 @@
+# Release 401 (26 Oct 2022)
+
+## General
+
+* Add support for using path-style access for all requests to S3 when using
+  fault-tolerant execution with exchange spooling. This can be enabled with the
+  `exchange.s3.path-style-access` configuration property. ({issue}`14655`)
+* Add support for polymorphic table functions in file-based access control. ({issue}`13713`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-401.md:8]: I would just say "table functions".

File: docs/src/main/sphinx/release/release-401.md [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+* Add distribution data for output buffer utilization to
```
Comment (martint) [inline docs/src/main/sphinx/release/release-401.md:9]: > Add output buffer utilization distribution to ...

File: docs/src/main/sphinx/release/release-401.md [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+  `EXPLAIN ANALYZE VERBOSE`. ({issue}`14596`)
+* Add distribution data for operator blocked time to `EXPLAIN ANALYZE VERBOSE`. ({issue}`14640`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-401.md:11]: > Add operator blocked time distribution ...

File: docs/src/main/sphinx/release/release-401.md [hunk 15]
```diff
@@ -0,0 +1,74 @@
...
+* Improve performance and reliability of `INSERT`, `MERGE`, and table optimization. ({issue}`14553`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-401.md:12]: What's "table optimization" ?

File: docs/src/main/sphinx/release/release-401.md (hunk 9 above)
Comment (martint) [reply docs/src/main/sphinx/release/release-401.md:15]: "the build side" is a term of art that most people won't understand. I would simplify to "for joins"

File: docs/src/main/sphinx/release/release-401.md [hunk 16]
```diff
@@ -0,0 +1,74 @@
...
+* Fix underestimation of CPU usage and scheduled time statistics for the build
+  side of joins in `EXPLAIN ANALYZE`. ({issue}`14572`)
+
+## Delta Lake connector
+
+* Add support for writing to tables with [Delta Laker writer protocol version 4](https://docs.delta.io/latest/versioning.html#features-by-protocol-version).
+  This does not yet include support for [change data feeds](https://docs.delta.io/2.0.0/delta-change-data-feed.html)
+  or generated columns. ({issue}`14573 `)
+* Add support for writes on Google Cloud Storage. ({issue}`12264`)
+* Fix bug where executing a `COMMENT` or `ALTER TABLE ... ADD COLUMN` statement
```
Comment (martint) [inline docs/src/main/sphinx/release/release-401.md:24]: > Avoid overwriting the reader and writer versions when executing a `COMMENT` or `ALTER TABLE ... ADD COLUMN`

File: docs/src/main/sphinx/release/release-401.md [hunk 17]
```diff
@@ -0,0 +1,74 @@
...
+  `avro_schema_literal`. ({issue}`14426`)
+* Fix potential query failure or incorrect results when reading from a table
+  with the `avro.schema.literal` Hive table property set. ({issue}`14426`)
+* Fix failure when listing tables from the Glue metastore and one of the tables
+  has no properties. ({issue}`14577`)
+
+## Iceberg connector
+
+* Improve performance of the `remove_orphan_files` table procedure. ({issue}`13691`)
+* Fix query failure when analyzing a table containing a column with a
```
Comment (martint) [inline docs/src/main/sphinx/release/release-401.md:42]: > ... a table *that contains* a column

File: docs/src/main/sphinx/release/release-401.md [hunk 18]
```diff
@@ -0,0 +1,81 @@
...
+* Add `query` table function for query pass-through to the connector. ({issue}`14535`)
+
+## MySQL connector
+
+* Add support for writes when [fault-tolerant
+  execution](#fault-tolerant-execution) is enabled. ({issue}`14445`)
+
+## Pinot connector
+
+* Fix failure when executing a `SHOW CREATE TABLE` statement. ({issue}`14071`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-401.md:67]: > ... when executing `SHOW CREATE TABLE`.

**Comment by kokosing** [comment]
I am sorry to be late in the party. We should also mention https://github.com/trinodb/trino/pull/14774. That now it is possible to control access to polymorphic functions with file based access controls.

**Pull request merged**